lib/mayaUsd/fileio/utils/meshWriteUtils.cpp

```diff
@@ -980,7 +990,7 @@ bool UsdMayaMeshWriteUtils::writeUVSetsAsVec2fPrimvars(
             valueWriter);
 
         // Save the original name for roundtripping:
-        if (primVar) {
+        if (primVar && !preserveSetNames) {
```

Comment:
Since 'setName' may also be changed by uvSetRemaps, a better condition here would be: if (primVar && (setName != uvSetNames[i]))

Reply:
Great catch! Thanks @vlasovi!